templates/support/URLSession+Api.swift.hbs

```diff
@@ -8,12 +8,27 @@ extension Foundation.URLSession {
 
     static var apiSession = Foundation.URLSession(configuration: .default)
 
+    @available(*, renamed: "handleApiRequest(_:)")
     static func handleApiRequest(_ request: Foundation.URLRequest, completion: @escaping ((_ response: Foundation.HTTPURLResponse?, _ data: Data?, _ error: Error?) -> Void)) {
-
-        apiSession.dataTask(with: request) { data, response, error in
-            completion(response as? Foundation.HTTPURLResponse, data, error)
-        }.resume()
-
+        Task {
+            do {
+                let result = try await handleApiRequest(request)
+                completion(result.response, result.data, nil)
+            } catch {
+                completion(nil, nil, error)
+            }
+        }
+    }
+    
+    static func handleApiRequest(_ request: Foundation.URLRequest) async throws -> (response: HTTPURLResponse, data: Data) {
+        do {
+            let (data, response) = try await apiSession.data(for: request)
+            guard let httpResponse = response as? HTTPURLResponse else {
+                fatalError("Expected non-nil HTTPURLResponse from URLRequest")
```

Comment:
This is a little funny in a library like this? We've got an opportunity to report an error so perhaps we can report a `systemError` with a message or something like that?